ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/NIUtils.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api;
+
+import org.netbeans.api.annotations.common.CheckForNull;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.nativeimage.spi.NIConfigurationProvider;
+import org.openide.filesystems.FileObject;
+
+/**
+ * Native image utilities.
+ */
+public final class NIUtils {
```

Comment:
I suggest to merge `NIUtils` and `NIActions` into one class.

When at it I also don't like the `NI` prefix in the class names. I won't mind using `ni` or `svm` in the package name - however the classnames should rather use full name, imho. I'd use `NativeImage` instead of `NI` everywhere. I'd use `NativeImages` for the name of the class created by merging the utils & actions.

Reply:
Removed for now, we'll add it back together with projects-related changes.